src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeExtensions.cs

```diff
@@ -44,6 +37,11 @@ private static bool MatchesArgumentTypes(this MethodInfo mi, Type[] argTypes)
             Debug.Assert(mi != null);
```

Comment:
@JonHanna thanks. doing the changes. How did you figure out that, that was the problem in the 6 failing checks... will be a learning for me...